soc/nordic_nrf/nrf54h/arm/Kconfig.soc

```diff
@@ -0,0 +1,35 @@
+# Nordic Semiconductor nRF54H MCU line
+
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_NRF54HX
+	bool
+	select SOC_FAMILY_NRF
+	help
+	  Nordic Semiconductor nRF54H series MCU
```

Comment:
if we preserve the arch folder split under soc/nordic_nrf, we'll need to keep duplicating stuff like that, which to me, seems unnecessary. IMHO, we should have all CPUs defined in a single `Kconfig.soc`, regardless of their arch.

Reply:
Have moved the Kconfig files to the nrf54h directory, the directory split is still needed for arm and risc-specific files e.g. soc.c

Reply:
why don't we simply have `soc_arm.c` and `soc_riscv.c`? As said, the directory split and implied Kconfig duplication just feels wrong. Moving to a hwmv2 where there are arch folders within vendor soc folder isn't a great model.

Reply:
What file would `#include <soc.h>` which is used all over the place include?

Reply:
Nordic RISC-V code doesn't have `soc.h` since it's fortunately exempt from the [CMSIS mess](https://github.com/zephyrproject-rtos/zephyr/issues/57246). In fact, nobody should be using `soc.h` now that there's `cmsis_core.h` including it. The whole soc.h idea is so brain-damaging.

Reply:
they should simply have a single soc.h, it's now ignored by RISC-V arch code see e7cc2fafb49c1b760bdd85ebf2bc67c53b30c031. If they have unnecessary stuff in soc.h, they should just put it somewhere else. soc.h should just exist to include the "CMSIS glue code" (the entire HAL in most cases)

Reply:
Btw, the whole soc.h scheme is so messed up that stuff like https://github.com/zephyrproject-rtos/zephyr/commit/00566d64fa901cbd7725068f6ff82ee0f0fffc16 is needed because soc.h pollutes almost every single compilation unit.

Reply:
then you would still have all these files: https://github.com/tejlmand/zephyr/tree/nrf52_adopt_soc_scheme_v2/soc/nordic_nrf/nrf54h/riscv/common/vpr

Renaming them all to `riscv_<filename>` would look like a mess

Reply:
These files should be under `soc/nordic_nrf/common/vpr`, since the VPR core is present in eg nRF54L15 (not added yet). In the future, we should think about supporting RISC-V core setups under `arch/riscv/cores` or something similar, we could have scenarios where a certain RISC-V core implementation is supported by multiple different vendor SoCs.